python-avd/pyavd/_eos_designs/structured_config/network_services/vlans.py

```diff
@@ -54,9 +55,33 @@ def vlans(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
                 )
                 self.structured_config.vlans.append(vlan, ignore_fields=("tenant",))
 
+            all_primary_vlans = []
             # L2 Vlans per Tenant
             for l2vlan in tenant.l2vlans:
-                self.structured_config.vlans.append(self._get_vlan_config(l2vlan, tenant), ignore_fields=("tenant",))
+                if l2vlan.private_vlan:
+                    if not self.shared_utils.platform_settings.feature_support.private_vlan:
+                        msg = (
+                            "The primary VLAN feature is not enabled by default for this platform."
+                            " It can be enabled in the platform settings, but might need additional configurations to work, see the TOI at"
+                            " https://www.arista.com/en/support/toi/eos-4-25-0f/14609-support-for-private-vlan."
+                        )
+                        raise AristaAvdInvalidInputsError(msg)
+
+                    all_primary_vlans.append(l2vlan.private_vlan.primary_vlan)
+                    vlan = self._get_vlan_config(l2vlan, tenant)
+                    vlan.private_vlan = l2vlan.private_vlan
```

Comment:
Required update if you change the model like I suggested in the schema.
```suggestion
                    vlan.private_vlan._update(type=l2vlan.private_vlan.type, primary_vlan=l2vlan.private_vlan.primary_vlan
```